index.js

```diff
@@ -14,16 +22,229 @@ module.exports = postcss.plugin('alter-color', options => {
 	const initialColor = parseColor(options.from);
 	const finalColor = parseColor(options.to);
 
-	console.log(initialColor);
-	console.log(finalColor);
-
 	return (root, result) => {
 		root.walkRules(rule => {
 			rule.walkDecls(decl => {
 				if (canContainColor(decl.prop)) {
 					const parsedValue = cssTree.parse(decl.value, {context: 'value'});
 
-					console.log(JSON.stringify(parsedValue));
+					cssTree.walk(parsedValue, {
+						visit: 'Identifier',
+						enter(node) {
+							if (node.name === initialColor.keyword) {
+								node.name = finalColor.keyword;
+							}
+						}
+					});
+
+					cssTree.walk(parsedValue, {
+						visit: 'HexColor',
+						enter(node) {
+							if (node.value.length > 3) {
+								if (node.value === removeHashSymbol(initialColor.hex)) {
+									node.value = removeHashSymbol(finalColor.hex);
+								}
+							} else {
+								if (node.value === convertToShortHex(removeHashSymbol(initialColor.hex))) {
+									node.value = convertToShortHex(removeHashSymbol(finalColor.hex));
+								}
+							}
+						}
+					});
+
+					cssTree.walk(parsedValue, {
+						visit: 'Function',
+						enter(node) {
+
+							const ast = cssTree.toPlainObject(node);
+
+							const colorGenerate = ast.children.reduce((acc, item) => {
```

Comment:
It‘s not clear what does this mean. `colorGenerate` doesn’t make sense. Why are you join these parts again?

Aside… I may guess, you are using `cssTree.toPlainObject` to access array methods like `.map()` and `.reduce()`. Can you confirm that there are no such methods on original object, please?